CHANGELOG.md

```diff
@@ -0,0 +1,63 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+## [Unreleased]
+
+## [0.4.0] - 2020-03-15
+
+### Added
+#### General
+ - Symmetric encryption. (Rust, C#, JS)
+ - Derive key pair from a password. (Rust, C#, JS)
+ - Shamir Secret Sharing. (Rust, JS)
+ - Officialize Javascript/Typescript support.
+
+#### C#
+ - Documented most functions and enums.
+ - The following functions have been added in the ```Utils```. It is not recommended to use them for performance reasons. Keep using ```ValidateSignature(byte[], DataType)``` when possible.
+   - ```ValidateSignatureFromBase64```
+   - ```ValidateSignatureFromStream```
+
+### Changed
+#### C#
+ - Renamed functions to be more consistent with what they do. The old functions now have the *[Obsolete]* attribute. 
+   - ```StringToByteArray``` is now ```StringToUtf8ByteArray```
+   - ```ByteArrayToString``` is now ```ByteArrayToUtf8String```
+   - ```Decode``` is now ```DecodeFromBase64```
+   - ```Encode``` is now ```EncodeToBase64String```
+ - Fixed issue in the ```DeriveKey()``` function
+   - Confusion between bits and bytes. The keys generated by the function would be 256 byte instead of 32 byte. The error is now fixed and our decryption will fallback to 256 bytes if the decryption fail.
+ - ```DevolutionsCryptoExceptions``` have now more details in the message and the stack trace.
+
+### Breaking Changes
+#### C#
+ - Renamed the following function
+   - ```GenerateKeyExchange``` to ```GenerateKeyPair```
+ - Renamed the following class
+   - ```KeyExchange``` to ```KeyPair```
+ - The function ```DecryptWithKey``` had a ```iterations``` parameter that wasn't used so it has been removed.
```

Comment:
Don't forget to talk about the Length default that was changed from 256 to 32 in DeriveKey